ush/retrieve_data.py

```diff
@@ -289,6 +289,13 @@ def get_file_templates(cla, known_data_info, data_store, use_cla_tmpl=False):
     """
 
     file_templates = known_data_info.get(data_store, {}).get("file_names")
+
+    # Remove sfc files from fcst in file_names of FV3GFS for LBCs
+    # sfc files needed in fcst when time_offset is not zero.
+    if cla.external_model == "FV3GFS" and cla.ics_or_lbcs == "LBCS":
+        del file_templates['nemsio']['fcst'][1]
+        del file_templates['netcdf']['fcst'][1]
+
```

Comment:
I think you may need to add a similar entry to the "nemsio" format for this to work. Changing the format to `nemsio` for the new test case you added I get this error:
```
Traceback (most recent call last):
  File "/scratch2/BMC/gsd-hpcs/Daniel.Abdi/ufs-srweather-app/ush/retrieve_data.py", line 1012, in <module>
    main(sys.argv[1:])
  File "/scratch2/BMC/gsd-hpcs/Daniel.Abdi/ufs-srweather-app/ush/retrieve_data.py", line 814, in main 
    file_templates = get_file_templates(
  File "/scratch2/BMC/gsd-hpcs/Daniel.Abdi/ufs-srweather-app/ush/retrieve_data.py", line 296, in get_file_templates
    del file_templates['nemsio']['fcst'][1]
IndexError: list assignment index out of range

```

Reply:
I think this is happening because the `gfs_file_names` anchor is shared for all sources, so when it tries hpss and doesn't succeed it deletes the sfc entry. Then when it tries aws, the entry is already gone. I think you can solve this by not using anchors in the yaml file (bad) or by making a deep copy of the dictionary before deleting the entries.

Reply:
@danielabdi-noaa, nemsio is not available for the date. netcdf and grib2 are only available. Do we need to any other conditions for this?

Reply:
@danielabdi-noaa, I've added if-statements to check the availability of `netcdf` and `nemsio` for the cycle date in `jobs/JREGIONAL_GET_EXTRN_MDL_FILES`. If nemsio (or netcdf) is not available, the `get_extrn_ics/lbcs` will fail with an error message.

Reply:
@chan-hoo I still think we need a safeguard against multiple deletions of the `sfc` file -- best would be to find another way that does not delete entries in the `data_locations` directory. If `hpss` does not have the file for some reason (maybe it is down), and we want to try `aws` next, it will fail because the `sfc` entry has been deleted by HPSS. The invalid nemsio date (although wrong) shows what could happen if we can't find the file on hpss but could be available on aws. I added this before the `del` if statement to make it try `aws` successfully when it could not find it in `hpss`
```
file_templates = deepcopy(file_templates)
```

Reply:
@danielabdi-noaa, yes, I agree with you. Can you help me? I've sent an invitation to you.

Reply:
@chan-hoo No big deal, i've pushed the change i think will avoid multiple deletions of the `sfc` file. It is not an ideal solution so lets wait for @christinaholtNOAA .

Reply:
Thank you!